File: 0148-sort-list/0148-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        int n=0;
        ListNode temp=head;
        while(temp!=null){
            n++;
            temp=temp.next;
        }
        int a[]=new int[n];
        temp=head;
        int i=0;
         while(temp!=null){
            a[i++]=temp.val;
            temp=temp.next;
        }
        Arrays.sort(a);
        temp=head;
        i=0;
        while(temp!=null){
            temp.val=a[i++];
            temp=temp.next;
        }
        return head;
    }
}

File: 0148-sort-list/0148-sort-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def sortList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        temp=head
        l=[]
        while temp:
            l.append(temp.val)
            temp=temp.next
        l.sort()
        temp=head
        i=0
        while temp:
            temp.val=l[i]
            temp=temp.next
            i+=1
        return head


        
        